scripts/run_all_benchmarks.sh

```diff
@@ -19,44 +19,59 @@
 #
 # Should be run on a reference machine to gain accurate benchmarks
 
-# shellcheck disable=SC2181
+set -eEuo pipefail
 
-ECHO_CMD=${ECHO_CMD:="false"}
+ECHO_CMD="${ECHO_CMD:-false}"
+[ "${ECHO_CMD}" = "true" ] && set -x
 
-if [ "${ECHO_CMD}" == "true" ]; 
-then
-      set -x
-fi
-
-USE_DOCKER_IMAGE=${USE_DOCKER_IMAGE:=""}
-# Force specific set of pallets (spaced string)
-USE_PALLETS=${USE_PALLETS:=""}
-ENABLE_OVERHEAD=${ENABLE_OVERHEAD:="true"}
-ENABLE_MACHINE=${ENABLE_MACHINE:="true"}
+# The following line ensures we know the project root
+SOURCE_ROOT="${SOURCE_ROOT:-$(git rev-parse --show-toplevel)}"
 
-# The following line ensure we know the project root
-SOURCE_ROOT=${SOURCE_ROOT:-$(git rev-parse --show-toplevel)}
+# shellcheck source=/dev/null
+source "${SOURCE_ROOT}/scripts/bench_cfg.sh"
 
-. "${SOURCE_ROOT}/scripts/bench_cfg.sh"
+USE_DOCKER="${USE_DOCKER:-false}"
+ENABLE_PALLETS="${ENABLE_PALLETS:-false}"
+ENABLE_OVERHEAD="${ENABLE_OVERHEAD:-true}"
+ENABLE_MACHINE="${ENABLE_MACHINE:-true}"
+BENCH_BASE_PATH="${BENCH_BASE_PATH:-}"
+BASE_PATH_ARG=""
+[ -n "${BENCH_BASE_PATH:-}" ] && BASE_PATH_ARG="--base-path=${BENCH_BASE_PATH}"
 
-BENCH_SH=${BENCH_SH:-"${SOURCE_ROOT}/scripts/bench.sh"}
+BENCH_SH="${BENCH_SH:-${SOURCE_ROOT}/scripts/bench.sh}"
 # Define the error file.
-ERR_FILE=${ERR_FILE:-"benchmarking_errors.txt"}
+ERR_FILE="${ERR_FILE:-${SOURCE_ROOT}/benchmarking_errors.txt}"
 
-if [ -z "${USE_DOCKER_IMAGE}" ]; 
-then
-echo "[+] Compiling nh-node benchmarks..."
-cargo build \
-  --locked \
-  --features=runtime-benchmarks \
-  --bin nh-node
+if [ "${USE_DOCKER}" = "false" ]; then
+  echo "[+] Compiling nh-node benchmarks..."
+  cargo build \
```

Comment:
I know it was already this way before current PR, but I was wondering if we want to build with profile `production` here, since we are using `PROFILE=production` in the Docker case.
Something like:
  cargo build \
    --locked \
    --features=runtime-benchmarks \
    --profile production \
    --bin nh-node

  # The executable to use.
  NH_NODE="${PROJECT_ROOT}/target/release/nh-node"